### ISSUE TITLE
Implement a wellness index

### ISSUE THREAD

**Issue by ezio-melotti** [overthesun/simoc#142]
*This is copied from the thread `Future features for SIMOC` from 2020-11-01*

-----

Even though humans currently don't do much and their conditions don't affect their performances, we can still track their "wellness".  This can be done at two levels:
  1. tracking different aspects of the wellness (including but not necessarily limited to: tiredness/fatigue/sleep, hunger/(mal)nutrition, thirst/hydration, O2/CO2 levels);
  2. a general index that summarizes with a single number (e.g. from 1 to 100) the wellness of the individual (possibly averaging the other values);

Wellness is affected by two categories of values:
 1. things that are beneficial but missing; 
 2. things that are harmful but overabundant.

In the first category we have food, water, sleep, O2, etc.; in the second CO2, other gases, radiations, etc..  The values in the first category can be tied to the handling and calculation of deprivation -- in fact we are already doing it for food, water, and O2 but not for sleep (even though nothing currently prevents humans from sleeping).  Something similar can be done for the values in the second category (question: how do we handle a short burst of high-level radiation or CO2, compared to a low-level but prolonged exposition?).  In the future we might also want to implement something like **"stamina"**, that will be consumed as the humans work, and replenished when they rest/eat.  These are not unlike systems found in video games.

If we bring this idea one step further and generalize it a bit, we can apply the same concept to other agents, starting from plants.  For plants, the beneficial values are presence of light, water, nutrients, CO2. (Question: as far as I understand plants perform both photosynthesis and cellular respiration, the former during the day and the latter constantly, but more prominently when light is not available.  Is this correct?  Do we take cellular respiration into account in SIMOC?).  The wellness of the plant can therefore be calculated based on the amount of light/nutrients/CO2 it receives.

Machines could also have their wellness index.  For simple machines, it could be a fixed 100 (unless we start calculating wear and tear), for other machines, lack of e.g. electricity or water might create issues and malfunctioning, or an abundance of certain elements might create other problems (e.g. calcium buildup).

In summary each agent could define things that are beneficial or harmful, with different levels (e.g. a little too much CO2 and the human feels dizzy, but add more and they will die).  These will determine the overall well-being of the agent.
